### PR TITLE
feat: new blog post - the last dance

### PR DESCRIPTION
**Pull request by shaimendel**
New blog post about The Last Dance series!

Please review only docs/md/docs/2020-07-09-last-dance.md, can be viewed properly at https://github.com/shaimendel/dev-blog/blob/blog/last_dance/docs/md/docs/2020-07-09-last-dance.md

### CODE REVIEW

File: docs/md/docs/2020-07-09-last-dance.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
+# The Last (Engineering) Dance
+
+Watching `The Last Dance` was an incredible experience, much more than watching Michael Jordan or specifically to sports - I treated this series as a team building tutorial.<br>
+So many team building and personal growth topics were raised there, I just found myself sitting after every episode and thinking on the episode's relevancy to my work as team lead and what I can learn from it.
+
+![The Chicago Bulls](/docs/assets/last-dance/group-header.jpeg)
+
+## tl;dr
+
+This blog post is about taking lessons from `The Last Dance` and applying them on the engineering world. <br>
```
Comment (nirfuchs) [inline docs/md/docs/2020-07-09-last-dance.md]: > applying them `on`

replace with`to`

File: docs/md/docs/2020-07-09-last-dance.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+The coach/team lead has so much on their heads:
+* Satisfying all stakeholders inside the organization - bringing money in by winning/selling
+* Satisfying the fans/customers in the long term
+* Satisfying the team itself - personal growth, ego, salaries
+* Satisfying themselves - let's not forget that the coach/team lead wants to develop and enjoy as well
+<br>
+Let's compare the two approaches of the two coaches in the series:
+
+### Doug Collins - win by maximizing reliance on Michael Jordan
+Doug Collins was a coach that brought great results to the team by putting all the focus on Michael Jordan, all the game moves and tactics was built on Michael Jordan's abilities.
```
Comment (nirfuchs) [inline docs/md/docs/2020-07-09-last-dance.md]: > all the game moves and tactics `was` built

replace with `were`

File: docs/md/docs/2020-07-09-last-dance.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+* The coach (Phil Jackson and Doug Collins before him) - the team lead
+* Michael Jordan - a senior engineer
+* The rest of the Chicago Bulls team - engineers in the team
+* Jerry Reinsdorf and the rest of the Chicago Bulls mgmt - different stakeholders, "upper management"
+* The fans and the crowd - customers
+
+Now that we have this comparison in mind we can begin with the real fun :)
+
+## The team lead's approach to winning
+The coach/team lead has so much on their heads:
```
Comment (nirfuchs) [inline docs/md/docs/2020-07-09-last-dance.md]: > so much on their heads

replace with `a lot on their minds`

File: docs/md/docs/2020-07-09-last-dance.md [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+This made the Chicago Bulls a great team because they had one hell of a senior engineer :) but they could not win the championship.
+There are a lot of disadvantages to this approach:
+* The team's ability to deliver is equal to the senior engineers' ability to deliver
+* The ability to improve is equal to the ability of your specific senior engineers to improve
+* This is not scalable and relies too much on too few people in your team
+* The non-senior team members do not improve, this approach perpetuates the current "snapshot" of the team - the seniors will stay seniors, the non-seniors will not be promoted.
+
+So even though the results were good, they were not great. Remember all the concerns team leads have on their heads? this approach led to unsatisfied internal stakeholders (not winning the championship), unstaisfied fans/customers (not winning the championship => the product quality was limited and did not improve), unsatisfied team (they were all a shadow of the senior engineer with no future to be seniors themselves), which makes the team lead itself unsatisfied - and who can blame them, how can you be satisfied when everyone around you are not?
```
Comment (nirfuchs) [inline docs/md/docs/2020-07-09-last-dance.md]: > `unstaisfied` fans
replace with `unsatisfied`

> on their `heads`
replace with `minds`

> everyone around you `are` not
replace with `is`

File: docs/md/docs/2020-07-09-last-dance.md [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+### Doug Collins - win by maximizing reliance on Michael Jordan
+Doug Collins was a coach that brought great results to the team by putting all the focus on Michael Jordan, all the game moves and tactics was built on Michael Jordan's abilities.
+This is an example to a team lead that puts the weight of the team's ability to deliver only on the senior engineers in the team:
+* Tough and complicated tasks are assigned to the senior engineers
+* The non-senior engineers' purpose is to let the senior engineers run fast and deliver
+
+This made the Chicago Bulls a great team because they had one hell of a senior engineer :) but they could not win the championship.
+There are a lot of disadvantages to this approach:
+* The team's ability to deliver is equal to the senior engineers' ability to deliver
+* The ability to improve is equal to the ability of your specific senior engineers to improve
```
Comment (nirfuchs) [inline docs/md/docs/2020-07-09-last-dance.md]: should `your` be part of the sentence? the write-up does not mention any belonging to the reader itself

File: docs/md/docs/2020-07-09-last-dance.md [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+* Satisfying the fans/customers in the long term
+* Satisfying the team itself - personal growth, ego, salaries
+* Satisfying themselves - let's not forget that the coach/team lead wants to develop and enjoy as well
+<br>
+Let's compare the two approaches of the two coaches in the series:
+
+### Doug Collins - win by maximizing reliance on Michael Jordan
+Doug Collins was a coach that brought great results to the team by putting all the focus on Michael Jordan, all the game moves and tactics was built on Michael Jordan's abilities.
+This is an example to a team lead that puts the weight of the team's ability to deliver only on the senior engineers in the team:
+* Tough and complicated tasks are assigned to the senior engineers
```
Comment (nirfuchs) [inline docs/md/docs/2020-07-09-last-dance.md]: should `tough` be replaced with `difficult` or `interesting`?

File: docs/md/docs/2020-07-09-last-dance.md [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+* The non-senior engineers' purpose is to let the senior engineers run fast and deliver
```
Comment (nirfuchs) [inline docs/md/docs/2020-07-09-last-dance.md:38]: the rest of the engineers on the team are described differently on Line 19. for consistency sake it's worth to using the same terms in both lines.

File: docs/md/docs/2020-07-09-last-dance.md [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+
+This made the Chicago Bulls a great team because they had one hell of a senior engineer :) but they could not win the championship.
+There are a lot of disadvantages to this approach:
```
Comment (nirfuchs) [inline docs/md/docs/2020-07-09-last-dance.md:41]: this is a classic example of a high [bus factor ](https://en.wikipedia.org/wiki/Bus_factor)

File: docs/md/docs/2020-07-09-last-dance.md [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+* The ability to improve is equal to the ability of your specific senior engineers to improve
+* This is not scalable and relies too much on too few people in your team
+* The non-senior team members do not improve, this approach perpetuates the current "snapshot" of the team - the seniors will stay seniors, the non-seniors will not be promoted.
+
+So even though the results were good, they were not great. Remember all the concerns team leads have on their heads? this approach led to unsatisfied internal stakeholders (not winning the championship), unstaisfied fans/customers (not winning the championship => the product quality was limited and did not improve), unsatisfied team (they were all a shadow of the senior engineer with no future to be seniors themselves), which makes the team lead itself unsatisfied - and who can blame them, how can you be satisfied when everyone around you are not?
+
+### Phil Jackson - win by maximizing the team performance
+Phil was hard to process for the team, he arrived with a completely different approach which basically turned upside down the way the team behaved so far. <br>
```
Comment (nirfuchs) [inline docs/md/docs/2020-07-09-last-dance.md]: `basically` seems unnecessary in this sentence

File: docs/md/docs/2020-07-09-last-dance.md [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+The discussions and plans moved from "what can we do to make Michael Jordan score?" to "what can we do to win in the long term?". This is a huge difference that made the real change in the group. <br>
+Phil put emphasize on team building, made everyone focused on the target of winning and to rely on each other. "There is no I in team". <br><br>
```
Comment (nirfuchs) [inline docs/md/docs/2020-07-09-last-dance.md]: > Phil put `emphasize`
replace with `emphasis`

File: docs/md/docs/2020-07-09-last-dance.md [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+
+After some adaption time there was a huge difference in the team performace that led to winning the championship. All the disadvantages from before disappeared:
```
Comment (nirfuchs) [inline docs/md/docs/2020-07-09-last-dance.md]: > the team `performace`
replace with `performance`

File: docs/md/docs/2020-07-09-last-dance.md [hunk 12]
```diff
@@ -0,0 +1,96 @@
...
+* The team's ability to deliver is equal to the entire team's ability to deliver
+* The ability to improve is equal to the entire team's ability to improve
+* This is as scalable as it can get - it reliaes on everyone in the team
```
Comment (nirfuchs) [inline docs/md/docs/2020-07-09-last-dance.md]: > it `reliaes` on
replace with `relies`

File: docs/md/docs/2020-07-09-last-dance.md [hunk 13]
```diff
@@ -0,0 +1,96 @@
...
+* The non-senior team members improve because the senior team members push them to excellence
```
Comment (nirfuchs) [inline docs/md/docs/2020-07-09-last-dance.md:59]: The same comment from line 38 applies here.

> the rest of the engineers on the team are described differently on Line 19. for consistency sake it's worth using the same terms in both lines.

File: docs/md/docs/2020-07-09-last-dance.md [hunk 14]
```diff
@@ -0,0 +1,96 @@
...
+
+This approach makes the stakeholders, fans/customers and the team satisfied.<br>
```
Comment (nirfuchs) [inline docs/md/docs/2020-07-09-last-dance.md:61]: add comma after `fans/customers`

File: docs/md/docs/2020-07-09-last-dance.md [hunk 15]
```diff
@@ -0,0 +1,96 @@
...
+When you put weight on the long term goal and make sure everyone on the team improves you get the best outcomes. Focusing on one person or on short term goals (winning only the next game and not the next seasons) don't get you to your maximum.
```
Comment (nirfuchs) [inline docs/md/docs/2020-07-09-last-dance.md]: IMO worth adding that there are no short cuts if you wish to get to your maximum
> `don't` get you
replace with `doesn't`

File: docs/md/docs/2020-07-09-last-dance.md [hunk 16]
```diff
@@ -0,0 +1,96 @@
...
+
+![Michael Jordan and Dennis Rodman](/docs/assets/last-dance/mj-and-dennis-rodman.jpeg)
+
+## The role of the senior team member
```
Comment (nirfuchs) [inline docs/md/docs/2020-07-09-last-dance.md:66]: Should the term senior team member be replaced with the term senior engineer? the text above and below mentions the term senior engineer

File: docs/md/docs/2020-07-09-last-dance.md [hunk 17]
```diff
@@ -0,0 +1,96 @@
...
+Doug Collins believed that the senior engineer's role is to bring success by leading the team and carry it forward.<br>
```
Comment (nirfuchs) [inline docs/md/docs/2020-07-09-last-dance.md]: When writing `role` do you mean `responsibility`?

File: docs/md/docs/2020-07-09-last-dance.md [hunk 18]
```diff
@@ -0,0 +1,96 @@
...
+Phil Jackson believed that the senior engineer's role is to bring success by pushing the team to excellence "from behind", being a servant leader that defines success by the team's success and not by personal success.
```
Comment (nirfuchs) [inline docs/md/docs/2020-07-09-last-dance.md]: when writing "from behind" do you mean from "behind the scenes"?

File: docs/md/docs/2020-07-09-last-dance.md [hunk 19]
```diff
@@ -0,0 +1,96 @@
...
+
+This is a conversation that blew my mind, it shows so clearly the difference between the two approaches. The problem is they are both right :)
+It took time for Michael Jordan to understand that there is indeed an I in win, but it had a different meaning than what he used to think. <br>
+Michael Jordan was a critical senior engineer in the team's success:
+* He pushed the other team members to excellence during practices. He expected them more, pushed them to be better. He had high expectations of them
+* He pushed the other team members to excellence during games
+* He helped the team to stay focused when their spirit took a hit. Michael had the ability to stay focused and perform even when they were losing, and this approach led other to play accordingly
+
+During their time under Phil's leadership, the best time this group has ever had, Michael wasn't the only scorer, the game tactics did not focus on Michael, and it took him time to process it. There is really no I in team.<br>
+When Michael saw the results of this new approach he was convinced that there is definitely an I in win, but only as part of a team.
```
Comment (nirfuchs) [inline docs/md/docs/2020-07-09-last-dance.md]: should 

> an I in win

be put in quotes?

File: docs/md/docs/2020-07-09-last-dance.md [hunk 20]
```diff
@@ -0,0 +1,96 @@
...
+<br><br>
+The series demonstrated so well the most effective way we can define the senior engineer role:
+* The senior engineer's success criteria is the team's performance and not their personal performance
```
Comment (nirfuchs) [inline docs/md/docs/2020-07-09-last-dance.md]: - IMO success is something you have to maintain over time and the senior engineer is a key player in doing that

How about replacing 

> and not their personal performance

and `not just their individual performance`?

File: docs/md/docs/2020-07-09-last-dance.md [hunk 21]
```diff
@@ -0,0 +1,96 @@
...
+<br>
+> Phil: There is no I in team <br>
+> Michael: But there is an I in win <br>
+
+This is a conversation that blew my mind, it shows so clearly the difference between the two approaches. The problem is they are both right :)
+It took time for Michael Jordan to understand that there is indeed an I in win, but it had a different meaning than what he used to think. <br>
+Michael Jordan was a critical senior engineer in the team's success:
+* He pushed the other team members to excellence during practices. He expected them more, pushed them to be better. He had high expectations of them
+* He pushed the other team members to excellence during games
+* He helped the team to stay focused when their spirit took a hit. Michael had the ability to stay focused and perform even when they were losing, and this approach led other to play accordingly
```
Comment (nirfuchs) [inline docs/md/docs/2020-07-09-last-dance.md]: - replace `had the ability` with `could`

>  led `other` to
replace with `others`

File: docs/md/docs/2020-07-09-last-dance.md [hunk 22]
```diff
@@ -0,0 +1,96 @@
...
+
+During their time under Phil's leadership, the best time this group has ever had, Michael wasn't the only scorer, the game tactics did not focus on Michael, and it took him time to process it. There is really no I in team.<br>
+When Michael saw the results of this new approach he was convinced that there is definitely an I in win, but only as part of a team.
+<br><br>
+The series demonstrated so well the most effective way we can define the senior engineer role:
+* The senior engineer's success criteria is the team's performance and not their personal performance
+* Improve yourself by making others better - don't be frustrated if the there is a gap of knowledge between yourself and the team, see it as an opportunity of personal growth and have high expectations from your team mates
+
+## Living in a roller coaster
+Basketball/engineering team's life is like a roller coaster, you sometimes get super high after a win and sometime get depressed by a devastating loss. It is hard to create a strong team spirit like that, and I found the way the Chicago Bulls coped with it very interesting and we should all apply it to our engineering teams:
```
Comment (nirfuchs) [inline docs/md/docs/2020-07-09-last-dance.md]: > a win and `sometime` get

replace with `sometimes`

File: docs/md/docs/2020-07-09-last-dance.md [hunk 23]
```diff
@@ -0,0 +1,96 @@
...
+## The role of the senior team member
+Doug Collins believed that the senior engineer's role is to bring success by leading the team and carry it forward.<br>
+Phil Jackson believed that the senior engineer's role is to bring success by pushing the team to excellence "from behind", being a servant leader that defines success by the team's success and not by personal success.
+<br>
+> Phil: There is no I in team <br>
+> Michael: But there is an I in win <br>
+
+This is a conversation that blew my mind, it shows so clearly the difference between the two approaches. The problem is they are both right :)
+It took time for Michael Jordan to understand that there is indeed an I in win, but it had a different meaning than what he used to think. <br>
+Michael Jordan was a critical senior engineer in the team's success:
```
Comment (nirfuchs) [inline docs/md/docs/2020-07-09-last-dance.md:75]: would it make sense to say that throughout his career Michael Jordan had learned from his failures?
and made sure to be accountable for them? he made sure he would not repeat them nor his team members

File: docs/md/docs/2020-07-09-last-dance.md (hunk 10 above)
Comment (nirfuchs) [inline docs/md/docs/2020-07-09-last-dance.md]: Phil being an inclusive leader, fostering collaboration (reducing bus factor) and not necessarily hiring new team members for their resume but their ability to complement the team. letting their creativity loose

File: docs/md/docs/2020-07-09-last-dance.md (hunk 18 above)
Comment (shaimendel) [reply docs/md/docs/2020-07-09-last-dance.md]: nope, meant to "push" them from behind, encouraging them to improve and not doing the job for them, as explained later in the sentence

**Pull request merged**